internal/util/vault.go

```diff
@@ -208,12 +214,16 @@ func (kms *VaultKMS) SavePassphrase(key, value string) error {
 
 // DeletePassphrase deletes passphrase from Vault.
 func (kms *VaultKMS) DeletePassphrase(key string) error {
-	vaultToken, err := kms.getAccessToken()
-	if err != nil {
-		return fmt.Errorf("could not retrieve vault token to delete the passphrase at %s: %s",
-			key, err)
+	vaultToken := ""
```

Comment:
I think it would be cleaner to add this inside getAccessToken() method - if vaultToken auth is in use - just return it, if not - request the token by using k8s auth

Reply:
Actually thats what I did at first place. then the check looks something like this: ( which somehow I didnt like :)) 

```
 func (kms *VaultKMS) getAccessToken() (string, error) {
+       if kms.VaultToken != ""{
+               return kms.VaultToken, nil
+       }

```
but I can revert to that..

Reply:
it looks a lot like memoization solution often used, so I don't see why this wouldn't be the good thing to do :)

Reply:
Yeah.. different flavour of choice :).. reverting to first version.. thanks!

Reply:
@vasyl-purchel reverted to first model .. ptal.